src/book.py

```diff
@@ -180,39 +217,79 @@ def tokenize(self, on, rem_stopwords=True, stopword_lang='english',
                     str.maketrans('', '', string.punctuation + '”“’'))
         return token
 
-    def snippet(self, length, on, non_cont=False, inplace=False):
+    def snippet(self, length, on, groups=1, non_cont=False, randomized=True, rem_stopwords=False, ret_as_arr=False, random_seed=None, inplace=False):
         """
+        Returns snippets of char/words/sent of various length depending on input.
+        # of snippets = groups (Default=1). # of char/word/sent of snippets = length.
+        non-cont: (Default = False): If True, returns random char/word/sent from starting point
+        randomized: (Default = False): If True, chooses a random point to start making snippets
+        rem_stopwords: Same as in Tokenize
+        ret_as_array: (Default = False): Returns one BookText object by appending all groups of snippets, else returns
+                                         an array of BookText objects
+        random_seed: When positive, generates the same snippets everytime
+        
         """
-        # TODO add ability for random seed
+        if (random_seed is not None):
+            random.seed(random_seed)
+            
         if 'char' in on.lower():
-            assert length < len(self._text)
+            tokens = self._text
         elif 'word' in on.lower():
-            tokens = word_tokenize(self._text)
-            assert length < len(tokens)
+            tokens = self.tokenize('word', rem_stopwords)
         elif 'sent' in on.lower():
-            tokens = sent_tokenize(self._text)
+            tokens = self.tokenize('sent', rem_stopwords)
         else:
             raise KeyError(
-                "Arugument 'on' must refer to character, word, or sentence")
-
-        if non_cont:
-            # TODO add functionality for non-continuous random sampling
-            pass
+                "Argument 'on' must refer to character, word, or sentence")
+            
+        assert groups*length <= len(tokens)
+        return_array = []
+
+        if randomized:
+            start = random.randint(0, len(tokens) - groups*length)
+            tokens = tokens[start:]
+            start = 0
```

Comment:
`start` is set back to 0 here, when two lines above it is set as a random integer.

Reply:
because I am deleting everything in tokens upto the starting point. Therefore, start now coincides with the 1st element of tokens.

Reply:
I see

Reply:
Please add comments here so this is made clear